src/mlx90632_extended_meas.c

```diff
@@ -0,0 +1,333 @@
+    // temp variables
+    double KsTAtmp, Alpha_corr;
+    double Ha_customer, Hb_customer;
+
+    Ha_customer = Ha / ((double)16384.0);
+    Hb_customer = Hb / ((double)1024.0);
+    calcedGa = ((double)Ga * (prev_object_temp - 25)) / ((double)68719476736.0);
+    KsTAtmp = (double)Fb * (TAdut - 25);
+    calcedGb = KsTAtmp / ((double)68719476736.0);
+    Alpha_corr = (((double)(Fa * POW10)) * Ha_customer * (double)(1.0 + calcedGa + calcedGb)) /
+                 ((double)70368744177664.0);
+    calcedFa = object / (emissivity * (Alpha_corr / POW10));
+
+    first_sqrt = sqrt(calcedFa + TaTr4);
+
+    return sqrt(first_sqrt) - 273.15 - Hb_customer;
+}
+
+double mlx90632_calc_temp_object_extended(int32_t object, int32_t ambient, double reflected,
+                                          int32_t Ea, int32_t Eb, int32_t Ga, int32_t Fa, int32_t Fb,
+                                          int16_t Ha, int16_t Hb)
+{
+    double kEa, kEb, TAdut;
+    double temp = 25.0;
+    double tmp_emi = mlx90632_get_emissivity();
+    double TaTr4;
+    double ta4;
+    int8_t i;
+
+    kEa = ((double)Ea) / ((double)65536.0);
+    kEb = ((double)Eb) / ((double)256.0);
+    TAdut = (((double)ambient) - kEb) / kEa + 25;
+
+    TaTr4 = reflected + 273.15;
+    TaTr4 = TaTr4 * TaTr4;
+    TaTr4 = TaTr4 * TaTr4;
+    ta4 = TAdut + 273.15;
+    ta4 = ta4 * ta4;
+    ta4 = ta4 * ta4;
+    TaTr4 = TaTr4 - (TaTr4 - ta4) / tmp_emi;
+
+    //iterate through calculations
+    for (i = 0; i < 5; ++i)
+    {
+        temp = mlx90632_calc_temp_object_iteration_extended(temp, object, TAdut, TaTr4, Ga, Fa / 2, Fb, Ha, Hb, tmp_emi);
+    }
+
+    return temp;
+}
+
+
+int32_t mlx90632_set_meas_type(uint8_t type)
+{
+    int32_t ret;
+    uint16_t reg_ctrl;
+
+    if ((type != MLX90632_MTYP_MEDICAL) & (type != MLX90632_MTYP_EXTENDED))
+        return -EINVAL;
+
+    ret = mlx90632_i2c_write(0x3005, MLX90632_RESET_CMD);
+    if (ret < 0)
+        return ret;
+
+    ret = mlx90632_i2c_read(MLX90632_REG_CTRL, &reg_ctrl);
+    if (ret < 0)
+        return ret;
+
+    reg_ctrl = reg_ctrl & (~MLX90632_CFG_MTYP_MASK & ~MLX90632_CFG_PWR_MASK);
+    reg_ctrl |= (MLX90632_MTYP_STATUS(type) | MLX90632_PWR_STATUS_HALT);
+
+    ret = mlx90632_i2c_write(MLX90632_REG_CTRL, reg_ctrl);
+    if (ret < 0)
+        return ret;
+
+    ret = mlx90632_i2c_read(MLX90632_REG_CTRL, &reg_ctrl);
+    if (ret < 0)
+        return ret;
+
+    reg_ctrl = reg_ctrl & ~MLX90632_CFG_PWR_MASK;
+    reg_ctrl |= MLX90632_PWR_STATUS_CONTINUOUS;
+
+    ret = mlx90632_i2c_write(MLX90632_REG_CTRL, reg_ctrl);
+
+    return ret;
+}
+
+int32_t mlx90632_get_meas_type(void)
+{
+    int32_t ret;
+    uint16_t reg_ctrl;
+
+    ret = mlx90632_i2c_read(MLX90632_REG_CTRL, &reg_ctrl);
+    if (ret < 0)
+        return ret;
+
+    reg_ctrl = reg_ctrl & MLX90632_CFG_MTYP_MASK;
+    reg_ctrl = reg_ctrl >> 4;
+
+    if ((reg_ctrl != MLX90632_MTYP_MEDICAL) & (reg_ctrl != MLX90632_MTYP_EXTENDED))
```

Comment:
Remove line 125 and use status below to improve readability. I would also somehow include that mask, but thats maybe a stretch too far since you reuse it nicely.

```suggestion
    if ((reg_ctrl != MLX90632_MTYP_STATUS_MEDICAL) & (reg_ctrl != MLX90632_MTYP_STATUS_EXTENDED))
```

Reply:
nope, bad comment.